torch/fx/passes/reinplace.py

```diff
@@ -0,0 +1,257 @@
+import torch
+import torch.fx as fx
+from torch.fx._compatibility import compatibility
+
+from torch.multiprocessing.reductions import StorageWeakRef
+
+from collections import defaultdict
+from typing import Iterable
+import itertools
+
+__all__ = ['reinplace']
+
+def concatMap(func, xs):
+    for x in xs:
+        for r in func(x):
+            yield r
+
+class GatherReinplaceDatastructures(torch.fx.Interpreter):
+    def __init__(self, gm):
+        super().__init__(gm)
+        self.node_counter = -1
+        # graph variable names of all program inputs
+        self.input_names = []
+        # Groups sets of the variable names of all aliased variables in the program
+        self.alias_map = defaultdict(set)
+        # For a given variable, tells you the names of all nodes in the program that use it.
+        # (including the index of that node, so we know where in the program it is)
+        self.names_to_used_nodes = defaultdict(list)
```

Comment:
It's funny to see this because in TorchScript IR, uses information (what nodes use a node) is so important that it is baked into the IR representation and must be kept up to date haha. FX isn't like that though lol.

Reply:
list is pretty non-descriptive, so a type annotation or a comment saying how exactly the node + index is represented would help here